src/CoreWCF.Http/tests/ServiceAuthBehaviorTest.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.ServiceModel.Security;
+using System.Threading.Tasks;
+using CoreWCF.Configuration;
+using CoreWCF.Description;
+using CoreWCF.IdentityModel.Policy;
+using CoreWCF.Primitives.Tests.CustomSecurity;
+using Helpers;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.Http.Tests
+{
+    public class ServiceAuthBehaviorTest
+    {
+        private ITestOutputHelper _output;
+
+        public ServiceAuthBehaviorTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        public void BasicHttpRequestReplyEchoWithServiceBehavior()
+        {
+            string testString = new string('a', 3000);
+            var host = ServiceHelper.CreateWebHostBuilder<Startup>(_output).Build();
+            using (host)
+            {
+                host.Start();
+                var httpBinding = ClientHelper.GetBufferedModeBinding();
+                var factory = new System.ServiceModel.ChannelFactory<ClientContract.IEchoService>(httpBinding,
+                    new System.ServiceModel.EndpointAddress(new Uri("http://localhost:8080/BasicWcfService/basichttp.svc")));
+                var channel = factory.CreateChannel();
+                var result = channel.EchoString(testString);
+                Assert.Equal(testString, result);
+            }
+        }
+
+        [Fact]
+        public void AccessDeniedForBasicHttpRequestReplyEcho()
+        {
+            string testString = new string('a', 3000);
+            var host = ServiceHelper.CreateWebHostBuilder<Startup>(_output).Build();
+            using (host)
+            {
+                host.Start();
+                var httpBinding = ClientHelper.GetBufferedModeBinding();
+                var factory = new System.ServiceModel.ChannelFactory<ClientContract.IEchoService>(httpBinding,
+                    new System.ServiceModel.EndpointAddress(new Uri("http://localhost:8080/BasicWcfService/basichttp.svc")));
+                var channel = factory.CreateChannel();
+                SecurityAccessDeniedException ex = Assert.Throws<SecurityAccessDeniedException>(() => channel.EchoToFail(testString));
+                Assert.Equal("Access is denied.", ex.Message);
+            }
+        }
+
+        internal class Startup
+        {
+            public void ConfigureServices(IServiceCollection services)
+            {
+                services.AddServiceModelServices();
+                var authBehavior = new ServiceAuthorizationBehavior();
+                services.AddSingleton<IServiceBehavior>(authBehavior);
+                services.AddSingleton(authBehavior);
```

Comment:
Should we add this by default in `services.AddServiceModelServices()`? What do you think about enabling people to do `services.AddSingleton<ServiceAuthorizationManager, MyTestServiceAuthorizationManager>()` and have it auto-wire in to the ServiceAuthorizationBehavior?